### PR TITLE
Add additional stacktrace information for throwable assertions

### PR DESCRIPTION
**Pull request by benoitdupont**
Show the stack trace for the Throwable under test when assertions for Throwable hasMessageContaining and hasMessageContainingAll fails

#### Check List:
* Fixes #1355
* Unit tests : YES 
* Javadoc with a code example (on API only) : NA

### CODE REVIEW

**Comment by benoitdupont** [comment]
Hello, this is my first Open Source contribution, any feedback welcome !

Here are some questions and remarks to be considered before merging (maybe I should have done a draft PR instead ?):

* I did not add specific methods for theses signatures, as they are not used in the caller code:
  * `ShouldContainCharSequence.shouldContains(Throwable actual, CharSequence sequence, ComparisonStrategy comparisonStrategy)`
  * `ShouldContainCharSequence.shouldContains(Throwable actual, CharSequence sequence, Set<? extends CharSequence> notFound, ComparisonStrategy comparisonStrategy)`

* I don't know if you want to extract the added logic for `Throwable` from `ShouldContainCharSequence` in a specific class, that delegate then to `ShouldContainCharSequence` which is String specific, or if a small mix of Throwable and String concepts is acceptable in the same class.

* I added `ShouldContainThrowable_create_Test#should_create_error_message_with_escaping_percent` but not `ShouldContainThrowable_create_Test#should_create_error_message` without the escaping percent, I don't think it's pertinent to have both, unless ?

* `assertHasStackTraceContaining` still need to be adapted, it calls `ShouldContainCharSequence.shouldContains`, but there's some investigation needed as it doesn't work as-is.

* Can I migrate `Throwables_assertHasMessageContaining_Test.java` and `Throwables_assertHasMessageContainingAll_Test.java` (because I modified them for the original issue) to junit5 by reducing visibility of class and methods from public to package private in a separate commit, or we should avoid to do multiples things in a single PR ?

**Comment by joel-costigliola** [comment]
Thanks @benoitdupont , I will review your PR shortly!

File: src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+@DisplayName("ShouldContainThrowable create")
+class ShouldContainThrowable_create_Test {
+  @Test
+  void should_create_error_message_with_escaping_percent() {
+    // GIVEN
+    RuntimeException actual = new RuntimeException("You know nothing %");
+    // WHEN
+    String errorMessage = ShouldContainCharSequence.shouldContain(actual, "You know nothing % Jon Snow")
+                                                   .create(new TestDescription("TEST"));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java]: use static import om `shouldContain` to make the code a bit lighter

File: src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    // THEN
+    assertThat(errorMessage).isEqualTo("[TEST] %n" +
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java]: Replace `assertThat` by `then` from `BDDAssertions`.

File: src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                                       getStackTrace(actual));
+  }
+
+  @Test
+  void should_create_error_message_with_several_values_not_found() {
+    // GIVEN
+    RuntimeException actual = new RuntimeException("You know nothing");
+    // WHEN
+    String errorMessage = ShouldContainCharSequence.shouldContain(actual, array("You", "know", "nothing", "Jon", "Snow"),
+                                                                  newSet("Jon", "Snow"))
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java]: good test data :smile:

File: src/main/java/org/assertj/core/error/ShouldContainCharSequence.java [hunk 4]
```diff
@@ -39,6 +42,36 @@ public static ErrorMessageFactory shouldContain(CharSequence actual, CharSequenc
                                    StandardComparisonStrategy.instance());
   }
 
+  public static ErrorMessageFactory shouldContain(Throwable actual, CharSequence sequence) {
+    String format = "%n" +
+                    "Expecting:%n" +
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldContainCharSequence.java]: I would reword this to ` "Expecting throwable message:%n" ` as we know the following string is not any string but a throwable message

File: src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java (hunk 2 above)
Comment (benoitdupont) [reply src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java]: I just discovered BDDAssertions and saw that it's encouraged in the contributing guidelines. Is it to _allow to skip test execution when assumptions are not met_ as described in https://assertj.github.io/doc/#assertj-core-3.14.0-bdd-assumptions ?

File: src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java (hunk 3 above)
Comment (benoitdupont) [reply src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java]: Glad you like it ;-)

**Comment by benoitdupont** [comment]
Voilà :-) All remarks has been fixed, just one question remaining in the upper discussion. And I see that I forgot to rebase the latest commit, I'm battling with git now to do that.

**Comment by benoitdupont** [comment]
Ok seems good, next time I won't forget to rebase the upstream before

File: src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java (hunk 2 above)
Comment (joel-costigliola) [reply src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java]: yes but I'm not sure what the use case is here, can you clarify?

**Comment by joel-costigliola** [comment]
@benoitdupont  as long as you are the only one working on a branch you can do whatever git operations on it (like rewrite the history and force push).

File: src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java (hunk 2 above)
Comment (benoitdupont) [reply src/test/java/org/assertj/core/error/ShouldContainThrowable_create_Test.java]: What I meant is: should we use BDDAssertions when we use the _given_ method to be able to skip the test execution when assumptions are not met, or use BDDAssertions by default instead of _assertThat_ as a general coding guideline ?

**Comment by joel-costigliola** [comment]
Integrated thanks @benoitdupont!
